### PR TITLE
Portable Windows Builds

### LINKED ISSUE #343: windows portable release
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AntumDeluge**
- adds CMake option to compile with portability on Windows & package portable zip file
- adds portable zip file to releases workflow

Closes #343

### CODE REVIEW

**Comment by pktiuk** [comment]
I will soon test it and merge.  
For now everything looks fine

File: CMakeLists.txt [hunk 1]
```diff
@@ -585,6 +585,11 @@ elseif(WIN32)
     target_link_libraries(antimicrox
         ${WIN_LIBS}
     )
+
+    option(PORTABLE "Compile app for portability (Windows only)" OFF)
```
Comment (pktiuk) [inline CMakeLists.txt:589]: This option should be moved at the beginning of Cmake file (next to other options) and rename it to something like PORTABLE_WIN

```cmake
if(WIN32)
    option(PORTABLE_WIN "Compile app for portability" OFF)
endif(WIN32)
```

**Review by pktiuk — CHANGES_REQUESTED** [review]

**Comment by pktiuk** [comment]
@AntumDeluge

**Comment by AntumDeluge** [comment]
@pktiuk Done.

File: CMakeLists.txt (hunk 1 above)
Comment (AntumDeluge) [reply CMakeLists.txt]: Done.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -160,8 +160,16 @@ jobs:
             - name: Build Installer
               run: cmake --build '${{ github.workspace }}'/build --target package --parallel 8
 
+            - name: Configure CMake (portable)
+              run: cmake -DPORTABLE_WIN=ON -B '${{ github.workspace }}'/build
```
Comment (pktiuk) [inline .github/workflows/release.yml:164]: You forgot about some flags required by release
```
            - name: Configure CMake (portable)
              run: cmake -DCMAKE_BUILD_TYPE=Release -DCHECK_FOR_UPDATES=ON -DANTIMICROX_PKG_VERSION="GitHub Portable Windows Release" -DPORTABLE_WIN=ON -B '${{ github.workspace }}'/build
```

Comment (pktiuk) [inline .github/workflows/release.yml:164]:
```suggestion
              run: cmake -DCMAKE_BUILD_TYPE=Release -DCHECK_FOR_UPDATES=ON -DANTIMICROX_PKG_VERSION="GitHub Portable Windows Release" -DPORTABLE_WIN=ON -B '${{ github.workspace }}'/build
```

**Review by pktiuk — APPROVED** [review]

**Pull request merged**